tests/Tests/500__jupyterhub/test-jupyterlab-cpu-stresstest.robot

```diff
@@ -0,0 +1,119 @@
+*** Settings ***
+Resource         ../../Resources/ODS.robot
+Resource         ../../Resources/Common.robot
+Resource         ../../Resources/Page/ODH/JupyterHub/JupyterHubSpawner.robot
+Resource         ../../Resources/Page/ODH/JupyterHub/JupyterLabLauncher.robot
+Library          DebugLibrary
+
+#Suite Setup      Begin Web Test
+Suite Setup      Accelerated Setup Suite
+Suite Teardown   End Web Test
+
+
+*** Keywords ***
+Accelerated Setup Suite
+  Set Library Search Order  SeleniumLibrary
+
+  Open Browser  ${ODH_DASHBOARD_URL}  browser=${BROWSER.NAME}  options=${BROWSER.OPTIONS}
+
+
+*** Variables ***
+
+
+*** Test Cases ***
+
+Open RHODS Dashboard
+  [Tags]  Sanity
+  Login To RHODS Dashboard  ${TEST_USER.USERNAME}  ${TEST_USER.PASSWORD}  ${TEST_USER.AUTH_TYPE}
+  Wait for RHODS Dashboard to Load
+
+Can Launch Jupyterhub
+  [Tags]  Sanity
+  ${version-check} =  Is RHODS Version Greater Or Equal Than  1.4.0
+  IF  ${version-check}==True
+    Launch JupyterHub From RHODS Dashboard Link
+  ELSE
+    Launch JupyterHub From RHODS Dashboard Dropdown
+  END
+
+Can Login to Jupyterhub
+  [Tags]  Sanity
+  Login To Jupyterhub  ${TEST_USER.USERNAME}  ${TEST_USER.PASSWORD}  ${TEST_USER.AUTH_TYPE}
+  ${authorization_required} =  Is Service Account Authorization Required
+  Run Keyword If  ${authorization_required}  Authorize jupyterhub service account
+  Wait Until Page Contains Element  xpath://span[@id='jupyterhub-logo']
+
+Can Spawn Notebook
+  [Tags]  Sanity
+  ## I know the below is needed, but it's quite time consuming!
+  #Fix Spawner Status
+  Spawn Notebook With Arguments  image=s2i-generic-data-science-notebook
+
+Git Clone the notebooks we need
+  Wait for JupyterLab Splash Screen  timeout=60
+  Maybe Close Popup
+  ${is_launcher_selected} =  Run Keyword And Return Status  JupyterLab Launcher Tab Is Selected
+  Run Keyword If  not ${is_launcher_selected}  Open JupyterLab Launcher
+  Launch a new JupyterLab Document
+  Add and Run JupyterLab Code Cell in Active Notebook  !rm -rf ~/PublicNotebooks/
+  Close Other JupyterLab Tabs
+  Capture Page Screenshot
+  Navigate Home (Root folder) In JupyterLab Sidebar File Browser
+  Open With JupyterLab Menu  Git  Clone a Repository
+  Input Text  //div[.="Clone a repo"]/../div[contains(@class, "jp-Dialog-body")]//input  https://github.com/erwangranger/PublicNotebooks.git
```

Comment:
Notebooks needed by ods-ci are encouraged  to be in this repo: https://github.com/redhat-rhods-qe/ods-ci-notebooks-main

We already have very simple notebook that is used to stress the CPU:  https://github.com/redhat-rhods-qe/ods-ci-notebooks-main/blob/main/notebooks/200__monitor_and_manage/200__metrics/stress-cpu-all-cores-900s-100.ipynb

It is using https://pypi.org/project/cpu-load-generator, which has options to use 1 cpu, all of them or even create usage profiles

Reply:
Thanks. Great to know. That will help. 

As soon as I find the time, I'll cancel this PR and create new, more atomic ones, in the right projects.